scripts/deployTokens.ts

```diff
@@ -0,0 +1,126 @@
+import { starknet } from "hardhat";
+import {
+  StarknetContract,
+  StarknetContractFactory,
+  Account,
+} from "hardhat/types";
+import fs from "fs";
+
+/**
+ * deploys and initializes ETHStaticAToken on L2
+ * @param deployer the deployer starknet account
+ * @param name token's name
+ * @param symbol token's symbol
+ * @param decimals  token's symbol
+ * @param initial_supply oken's initial supply
+ * @param recipient mints initial supply to recipient
+ * @param controller the token controller/owner
+ * @param proxy_admin address of the proxy owner
+ */
+export async function deployETHStaticAToken(
+  deployer: Account,
+  name: string,
+  symbol: string,
+  decimals: bigint,
+  initial_supply: { low: bigint; high: bigint },
+  recipient: bigint,
+  controller: bigint,
+  proxy_admin: bigint
```

Comment:
why not replacing `proxy_admin` with the address of the `deployer`?

Reply:
the deployer is not necessarily the proxy admin

Reply:
but it's the case currently. my point is: no need to write a code too generic if we don't have a use-case for it yet.

Reply:
I get your point